scripts/match-2-yar.py

```diff
@@ -0,0 +1,868 @@
+ you may not use this file except in compliance with the License.
+You may obtain a copy of the License at: [package root]/LICENSE.txt
+Unless required by applicable law or agreed to in writing, software distributed under the License
+ is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and limitations under the License.
+
+match-2-yar
+
+Invoke capa to extract the capabilities of the given sample or list of samples,
+and emit the matches as yara rules.
+
+When providing multiple samples or directories the tool will attempt to create
+"super rules" based on overlapping signatures
+
+
+Example::
+
+    $ python scripts/match-2-yar.py /tmp/suspicious.dll_
+    ...
+
+Example::
+
+    $ python scripts/match-2-yar.py /tmp/suspicious.dll_ /tmp/suspicious2.dll_
+    ...
+
+"""
+import os
+import sys
+import json
+import logging
+import argparse
+import binascii
+import collections
+import multiprocessing
+import multiprocessing.pool
+from typing import Set, Dict, List
+from pathlib import Path
+from datetime import date
+
+import dnfile
+from envi.memcanvas import MemoryCanvas
+from dncil.clr.token import Token
+from vivisect.renderers import WorkspaceRenderer
+
+import capa.main
+import capa.rules
+import capa.engine
+import capa.helpers
+import capa.features
+import capa.exceptions
+import capa.render.utils as rutils
+import capa.render.verbose
+import capa.features.freeze
+import capa.render.result_document as rd
+from capa.features.common import OS_AUTO
+from capa.features.extractors.dnfile.extractor import DnfileFeatureExtractor
+
+try:
+    import yaramod
+    from mkyara import YaraGenerator
+    from capstone import CS_MODE_32, CS_MODE_64, CS_ARCH_X86, CS_OPT_SYNTAX_INTEL, Cs
+except ImportError:
+    print(
+        """\nFailed to import a module try installing required Python libraries with the following:
+pip install mkyara yaramod
+"""
+    )
+    sys.exit(1)
+
+
+logger = logging.getLogger("capa.match-2-yar")
+
+
+# Vivisect Related Classes and Functions
+
+
+class BufferCanvas(MemoryCanvas):
+    """Subclass of Vivisect Memory canvas that captures
+    disassemlby output as a string rather than printing to std.out
+    """
+
+    output = ""
+
+    def addText(self, text, tag=None):
+        """Overwriting the method responsible for writing to std.out"""
+        self.output += text
+
+
+def get_disassembly_output(vw, va, size):
+    """Get Vivisect's disassembly view for a given virtual addresss and size
+
+    Args:
+        vw: Vivisect Workspace
+        va: Virtual Address to start disassembling from
+        size: size in bytes to disassemble
+
+    Returns:
+        str: String containing vivisect's disassembly output
+    """
+    rend = WorkspaceRenderer(vw)
+    mcav = BufferCanvas(vw)
+    mcav.renderMemory(va, size, rend=rend)
+    return mcav.output
+
+
+def get_comment_for_func(vw, funcva):
```

Comment:
name could be clearer, e.g., `get_function_disassembly_output`?

Reply:
Updated